native/cpp/CommonCpp/CryptoTools/CryptoModule.cpp

```diff
@@ -374,33 +265,28 @@ void CryptoModule::removeSessionByDeviceId(const std::string &deviceId) {
   this->sessions.erase(deviceId);
 }
 
-OlmBuffer CryptoModule::pickleAccount(const std::string &secretKey) {
-  OlmAccount *olmAccount = this->getOlmAccount();
-  size_t accountPickleLength = ::olm_pickle_account_length(olmAccount);
-  OlmBuffer accountPickleBuffer(accountPickleLength);
-
-  size_t result = ::olm_pickle_account(
-      olmAccount,
-      secretKey.data(),
-      secretKey.size(),
-      accountPickleBuffer.data(),
-      accountPickleLength);
+std::string CryptoModule::pickleAccount(const std::string &secretKey) {
+  try {
+    // Convert secretKey to 32-byte array
+    std::array<uint8_t, 32> key;
+    std::copy_n(
+        secretKey.begin(), std::min(secretKey.size(), size_t(32)), key.begin());
 
-  if (accountPickleLength != result) {
-    throw std::runtime_error(
-        "Error in pickleAccount => " +
-        std::string(::olm_account_last_error(olmAccount)));
+    return std::string(this->vodozemacAccount->pickle(key));
+  } catch (const std::exception &e) {
+    throw std::runtime_error("error pickleAccount => " + std::string(e.what()));
   }
-  return accountPickleBuffer;
 }
 
 Persist CryptoModule::storeAsB64(const std::string &secretKey) {
   Persist persist;
-  persist.account = this->pickleAccount(secretKey);
+  std::string accountPickle = this->pickleAccount(secretKey);
+  persist.account = OlmBuffer(accountPickle.begin(), accountPickle.end());
 
   std::unordered_map<std::string, std::shared_ptr<Session>>::iterator it;
   for (it = this->sessions.begin(); it != this->sessions.end(); ++it) {
-    OlmBuffer buffer = it->second->storeAsB64(secretKey);
+    std::string sessionPickle = it->second->storeAsB64(secretKey);
+    OlmBuffer buffer(sessionPickle.begin(), sessionPickle.end());
```

Comment:
Why do we still need to use `OlmBuffer`? Is this type from inside of the old Olm library? Will it prevent us from removing that library?

Reply:
`OlmBuffer` is just an alias for `std::vector<std::uint8_t>`:
https://github.com/CommE2E/comm/blob/3b4513ac87073ea983ca278025f1e99efb23504a/native/cpp/CommonCpp/CryptoTools/Tools.h#L18